pkg/api/v1alpha2/types.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type DeschedulerPolicy struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// Profiles
+	Profiles []DeschedulerProfile `json:"profiles,omitempty"`
+
+	// NodeSelector for a set of nodes to operate over
+	NodeSelector *string `json:"nodeSelector,omitempty"`
+
+	// EvictLocalStoragePods allows pods using local storage to be evicted.
+	EvictLocalStoragePods *bool `json:"evictLocalStoragePods,omitempty"`
+
+	// EvictSystemCriticalPods allows eviction of pods of any priority (including Kubernetes system pods)
+	EvictSystemCriticalPods *bool `json:"evictSystemCriticalPods,omitempty"`
+
+	// IgnorePVCPods prevents pods with PVCs from being evicted.
+	IgnorePVCPods *bool `json:"ignorePvcPods,omitempty"`
+
+	// MaxNoOfPodsToEvictPerNode restricts maximum of pods to be evicted per node.
+	MaxNoOfPodsToEvictPerNode *int `json:"maxNoOfPodsToEvictPerNode,omitempty"`
+}
+
+type DeschedulerProfile struct {
+	// Name
+	Name string `json:"name,omitempty"`
+
+	// Enabled
+	Enabled *bool `json:"enabled,omitempty"`
```

Comment:
I don't think this needs to be a pointer. I believe the reason we made the `strategy.Enabled` a pointer originally was part of converting most of our fields to pointers, so that they could be omitted from yaml/json when encoded. But this is a required field, so it shouldn't ever be empty anyway

Reply:
Its a pointer to allow the flag to be omitted from the config. Do we need `enabled` always present in the config, even if its `true`?
IMO this flag can be made optional if its only for debugging. Otherwise it feels a repetition of flag across all strategies and profiles. wdyt ?

Reply:
+1 for not making it a pointer